agenthub/self_discover_agent/prompt.py

```diff
@@ -0,0 +1,214 @@
+
+from .agent_state_machine import SelfDiscoverState
+from .reasoning_action import REASONING_MODULE_LIST, ReasoningAction
+
+TASK_KEY = 'task'
+
+ROLE_DESCRIPTION = """You are a seasoned manager at a Fortune 500 company with a team of capable software engineers.
+Your responsibility is to plan the given task for your team. You must only plan and must not code yourself."""
+
+INTERACTION_SKILL = """You are working with a curious user that needs your help. You can directly interact with the user by replying with
+<execute_ask> and </execute_ask>. For example, <execute_ask> Are my assumptions correct? </execute_ask>.
+"""
+
+BROWSING_SKILL = """You can also browse the Internet with <execute_browse> and </execute_browse>.
+For example, <execute_browse> Tell me the usa's president using google search </execute_browse>.
+Or <execute_browse> Tell me what is in http://example.com </execute_browse>.
+"""
+
+SYSTEM_SUFFIX = """Let the following principles guide your response:
+- Before responding read all information carefully.
+- Take time to think.
+- Responses should be concise.
+- Use only information that's explicitly stated by the user.
+"""
+
+# TODO: allow agent to ask questions before specifying a plan
+# SYSTEM_MESSAGE = ROLE_DESCRIPTION + INTERACTION_SKILL + BROWSING_SKILL + SYSTEM_SUFFIX
+SYSTEM_MESSAGE = ROLE_DESCRIPTION + SYSTEM_SUFFIX
+
+SELECT_PROMPT = """Select several reasoning moduls that are crucial to solve the task provided by the user.
+
+## Task
+Recall from above the task: {task}
+
+## Reasoning modules list (0-based indexing)
+{reasoning_module_list}
+
+## Constraints
+Select between 3 and 5 task-relevant reasoning modules.
+
+Your response MUST be in JSON format. It must be an object with the field "{select_state_key}",
+which contains the indeces of selected reasoning modules. For example, for selecting reasoning modules
+"How could I devise an experiment to help solve that problem?" and "How could I measure progress on this problem?",
+you would reply with
+
+{{
+  "{select_state_key}": [0, 2]
+}}
+
+You MUST NOT include any other text besides the JSON response.
+"""
+
+ADAPT_PROMPT = """Rephrase and specify each previously selected reasoning module so that it better helps solving the task.
+
+## Selected reasoning modules
+{selected_reasoning_modules}
+
+## Task
+Recall from above the task is as follows: {task}
+
+## Constraints
+Your response MUST be in JSON format. It must be an object with the field "{adapt_state_key}", which contains key-value pairs describing the adapted reasoning modules.
+The key represents the title for the adapted reasoning module.
+For example,
+
+{{
+  "{adapt_state_key}": {{
+    "Title adapted reasoning module X": "Description adapted reasoning module X",
+    "Title adapted reasoning module Y": "Description adapted reasoning module Y",
+    }}
+}}
+
+You MUST NOT include any other text besides the JSON response.
+"""
+
+# TODO: modify example so that it is consistent with prompting
```

Comment:
Do we want to keep this commented out code?

Reply:
same here

Reply:
Okay, will remove the code for now and just leave the Todo comment